core/src/main/scala/kafka/admin/DeleteRecordsCommand.scala

```diff
@@ -22,31 +22,62 @@ import java.util.Properties
 
 import kafka.admin.AdminClient.DeleteRecordsResult
 import kafka.common.AdminCommandFailedException
-import kafka.utils.{CoreUtils, Json, CommandLineUtils}
+import kafka.utils.{CommandLineUtils, CoreUtils, Json}
 import org.apache.kafka.common.TopicPartition
 import org.apache.kafka.common.utils.Utils
 import org.apache.kafka.clients.CommonClientConfigs
 import joptsimple._
+import kafka.utils.json.JsonValue
 
 /**
  * A command for delete records of the given partitions down to the specified offset.
  */
 object DeleteRecordsCommand {
 
+  private[admin] val EarliestVersion = 1
+  private[admin] val supportedVersions = List(EarliestVersion)
+
   def main(args: Array[String]): Unit = {
     execute(args, System.out)
   }
 
   def parseOffsetJsonStringWithoutDedup(jsonData: String): Seq[(TopicPartition, Long)] = {
-    Json.parseFull(jsonData).toSeq.flatMap { js =>
-      js.asJsonObject.get("partitions").toSeq.flatMap { partitionsJs =>
-        partitionsJs.asJsonArray.iterator.map(_.asJsonObject).map { partitionJs =>
-          val topic = partitionJs("topic").to[String]
-          val partition = partitionJs("partition").to[Int]
-          val offset = partitionJs("offset").to[Long]
-          new TopicPartition(topic, partition) -> offset
-        }.toBuffer
+    Json.parseFull(jsonData) match {
+      case Some(js) => {
+
+        val version = js.asJsonObject.get("version") match {
+          case Some(jsonValue) => jsonValue.to[Int]
+          case None => EarliestVersion
+        }
+        if (!supportedVersions.contains(version))
+          throw new AdminOperationException(s"Not supported version field value $version")
+
+        parseJsonData(version, js)
+      }
+      case None => throw new AdminOperationException("Delete records data is empty")
+    }
+  }
+
+  def parseJsonData(version: Int, js: JsonValue): Seq[(TopicPartition, Long)] = {
+    version match {
+      case EarliestVersion => {
```

Comment:
I think the logic here is to parse data based on the supported version. In this case the supported version is 1. It does not really matter whether it is earliest version or not, right? This can be similar to org.apache.kafka.common.requests.ProduceRequest.requiredMagicForVersion.

Reply:
If I got this right the your point was why I was using the EarliestVersion and not simple 1. I was just trying to avoid a literal here because our first supported version is the "earliest" one but maybe in this case it makes much more sense.